cr.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  Netguru iOS code review task
+//
+
+class paymentViewController: UIViewController {
+    var delegate: PaymentViewControllerDelegate
+    let customView = PaymentView()
+    let payment: Payment?
+
+    func callbacks() {
+        view.backgroundColor = UIColor(displayP3Red: 1, green: 1, blue: 1, alpha: 1)
+        customView.didTapButton = {
+            if let payment = self.payment {
+                self.delegate.didFinishFlow(amount: payment.amount
+                                            currency: payment.currency)
+            }
+        }
+    }
+
+    func viewDidLoad() {
+        navigationController?.setNavigationBarHidden(false, animated: false)
+        customView.centerXAnchor.constraint(equalTo: view.centerXAnchor)
+        customView.centerYAnchor.constraint(equalTo: view.centerYAnchor)
+        view.addSubview(customView)
+        fetchPayment()
+        setupCallbacks()
+    }
+
+    internal let ViewModel: PaymentViewModel
+
+    func fetchPayment() {
+        customView.statusText = "Fetching data"
+        ApiClient.sharedInstance().fetchPayment { payment in
+            self.CustomView.isEuro = payment.currency == "EUR" ? true : false
+            if payment!.amount != 0 {
+                self.CustomView.label.text = "\(payment!.amount)"
```

Comment:
as mentioned in previous comments, `CustomView` will give compile time error. `payment!.amount` can occur runtime error in feature. it will better to define as `"\(payment?.amount ?? 0)"` or even better and correct way to check payment with `guard` or `if let...`:

```
ApiClient.sharedInstance.fetchPayment { [weak self] payment in
    guard let self = self, let payment = payment else { return }
    self.paymentView.isEuro = payment.currency.uppercased() == "EUR" ? true : false

    // I think this check is not correct and should be clarified with BA, SA, client, etc.
    // because in this case label can print any value except 0 (negative and positive).
    // If the values should be only positive, then check should be payment.amount > 0
    if payment.amount != 0 {
        self.paymentView.label.text = "\(payment.amount)"
    }
}
```